infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -40,47 +43,62 @@
  * In this scheme, the key for each value is also the name of the file where the value is stored.
  * The key is determined as (prefix + "_" + key).
  * The cache here serves mostly for easily managed synchronization of in-memory/file.
+ * Some on-disk files will be deleted when their number exceed user specified limit.
  *
  * <p>If 'opts' either has '--memory=true' or a log-path for storing files is not provided,
  * the store is just an in memory cache.
  *
  * <p>Created by mdhawan on 7/27/16.
  */
 
+@Slf4j
 public class DataStore implements IDataStore {
 
     static String EXTENSION = ".ds";
 
     @Getter
     private final Cache<String, Object> cache;
-    private final String logDir;
+    private final String logDirPath;
 
     @Getter
     private final long dsCacheSize = 1_000; // size bound for in-memory cache for dataStore
 
     private final boolean inMem;
 
+    private final int dsFileRetention;
+
+    // Prefixes of DataStore files that will be created on epoch change
+    private final Set<String> prefixesToClean;
+
     /**
      * Return a new DataStore object.
-     * @param opts  map of option strings
+     *
+     * @param opts map of option strings
+     * @param prefixes prefixes of files that need cleanup
      */
-    public DataStore(Map<String, Object> opts) {
+    public DataStore(@Nonnull Map<String, Object> opts,
```

Comment:
Instead of passing the prefixes to clean, I wonder if its "cleaner" to just pass a cleanup method instead of implementing it in this class?

Reply:
Done, thanks for the idea.